src/js/content_scripts/gamepad-navigator.js

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright (c) 2020 The Gamepad Navigator Authors
+See the AUTHORS.md file at the top-level directory of this distribution and at
+https://github.com/fluid-lab/gamepad-navigator/raw/master/AUTHORS.md.
+
+Licensed under the BSD 3-Clause License. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the BSD 3-Clause License at
+https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
+*/
+
+/* eslint-env browser */
+
+(function (fluid) {
+    "use strict";
+
+    var gamepad = fluid.registerNamespace("gamepad");
+
+    fluid.defaults("gamepad.gamepadNavigator", {
+        gradeNames: ["fluid.modelComponent"],
+        model: {
+            index: null,
+            axes: [],
+            buttons: []
+        },
+        listeners: {
+            "onCreate.gamepadListener": "{that}.gamepadListener"
+        },
+        frequency: 100,
+        connectivityIntervalReference: null,
+        invokers: {
+            gamepadListener: {
+                funcName: "gamepad.gamepadNavigator.gamepadListener",
+                args: ["{that}"]
+            },
+            transformArrayToObject: {
+                funcName: "gamepad.gamepadNavigator.arrayToObject",
+                args: ["{arguments}.0"]
+            }
+        }
+    });
+
+    /**
+     *
+     * Listens to the connected and disconnected events of the gamepad and updates the
+     * gamepad navigator component's model according to the state of the gamepad.
+     *
+     * @param {Object} that - The gamepad navigator component.
+     *
+     */
+    gamepad.gamepadNavigator.gamepadListener = function (that) {
```

Comment:
I think this should be split out into two functions, one to attach the listener to the window, the other to actually handle the event.  You could also argue for making a wrapper for all gamepad events and writing listeners for each type of event.  So you'd have an "attachListeners" function that ran on creation, and that adds a (window) event listener to fire the associated component event.  Then your "gamepadconnected" listener would be a normal Fluid component listener definition, which could be modified or replaced using configuration options.

Reply:
The second approach you suggested sounds quite interesting to me and I think providing the option to reconfigure would be good for other developers to extend the work as well as for maintaining the project in the long term. I am ready to proceed with the same if you allow me.